app/page.tsx

```diff
@@ -90,25 +82,49 @@ const HomeSearchPage: React.FC = () => {
     availableOnly: DEFAULT_AVAILABILITY_FILTER_OPTIONS,
     price_min: DEFAULT_MIN_PRICE,
     price_max: DEFAULT_MAX_PRICE,
-  });
+  };
+
+  const [searchParams, setSearchParams] = useState<searchParamsType | null>(null);
+
+  useEffect(() => {
+    const getInitialSearchParams = (): searchParamsType => {
+      const storedParams = localStorage.getItem("searchParams");
+      return storedParams ? JSON.parse(storedParams) : defaultSearchParams;
+    };
+
+    setSearchParams(getInitialSearchParams());
+  }, []);
+
+  useEffect(() => {
+    if (searchParams) {
+      localStorage.setItem("searchParams", JSON.stringify(searchParams));
+    }
+  }, [searchParams]);
 
-  // Handlers to update specific sections of searchParams
   const updateBookingInfoParams = (
-    newSearchParams: Partial<typeof searchParams>
-  ) => setSearchParams((prev) => ({ ...prev, ...newSearchParams }));
+    newSearchParams: Partial<searchParamsType>
+  ) => setSearchParams((prev) => ({ ...prev!, ...newSearchParams }));
+
+  const updateHotelSearchParams = (newHotelSearch: searchParamsType) => {
+    setSearchParams(newHotelSearch);
+  };
 
   useEffect(() => {
     if (!region) return;
     setSearchParams((prev) => ({
-      ...prev,
+      ...prev!,
```

Comment:
```suggestion
      ...prev,
```
If you make searchParams non-nullable, as suggested in another comment, you shouldn't need to assert the value exists.